samples/wifi/provisioning/implementation_details.rst

```diff
@@ -0,0 +1,268 @@
+The message shall be in type of Result defined in result.proto and encoded.
+The information of access point shall be in the scan_record field.
+In each sending function call, it should put only one scan result.
+
+STOP_SCAN procedure
+-------------------
+
+When request with Op Code of STOP_SCAN is received, the device shall send a command of stop scan to Wi-Fi subsystem, and depending on the result, a response message is sent to configurator.
+
+If the operation is successful, the device shall send response message with status of STATUS_SUCCESS.
+If internal error occurs, the status shall be STATUS_INTERNAL_ERROR.
+
+SET_CONFIG procedure
+--------------------
+
+When request with Op Code of SET_CONFIG is received, the device shall send a command of connecting to the Access Point specified in the config field to Wi-Fi subsystem, and depending on the result, a response message is sent to configurator.
+
+If the operation is successfully dispatched, the device shall send response message with status of STATUS_SUCCESS.
+If invalid argument is provided, the status shall be STATUS_INVALID_ARGUMENT.
+If internal error occurs, the status shall be STATUS_INTERNAL_ERROR.
+
+If the operation is successful, the device shall send state update to configurator.
+The message shall be in type of Result defined in result.proto and encoded.
+The updated state of Wi-Fi should be in state field. If the state is CONNECTION_FAILED, the device may fill reason field, whose value depends on the failure reason.
+If the state is CONNECTED, the device shall store the credentials in NVM.
+
+FORGET_CONFIG procedure
+-----------------------
+
+When request with Op Code of FORGET_CONFIG is received, the device shall remove stored credentials from NVM, and depending on the result, a response message is sent to configurator.
+
+If the operation is successful, the device shall send response message with status of STATUS_SUCCESS.
+And the device shall disconnect from access point if its state is connected.
+If internal error occurs, the status shall be STATUS_INTERNAL_ERROR.
+
+You may find the details of definition in .proto files under path of ./common/proto.
+
+Message delivery
+================
+
+To exchange messaged mentioned above, some sort of transport method is required.
+Besides, it should also provide a certain level of security.
+In our implementation, Bluetooth Low Energy (BLE) is chosen.
+It is widely deployed and available on most smartphones, which play the role of configurator.
+Meanwhile, we leverage its capability of information confidentiality and integrity.
+BLE itself has pairing function, and can provide different level of security.
+GATT allows us to assign different security level to different characteristic under the same service, making fine tune of access control possible.
+
+Service declaration
+-------------------
+
+The Wi-Fi Provisioning Service is instantiated as a Primary Service.
```

Comment:
What's a primary service?

It should probably be in lowercase letters.

Reply:
Primary service is a term defined by GATT.  Lowercase may be correct, after check Bluetooth specs.